[Howard Wong] iP

howardwhw2 opened the pull request:
# **Vanbot**


vanbot helps you organise your life better by:
- Helping you manage the things happening in your life
-  Easy to use

All you got to do is:
1. Download it from [here](https://github.com/howardwhw2/ip/releases/tag/A-Jar)
2. Open and run it
3.  Use it for **FREE** :smiley:

Features:
- [x] Managing tasks
- [ ] Adding GUI support (work in progress)
- [ ] Adding advertisments (coming soon)

> Download now while it is FREE

Here is some sample code:
```java
    public String getCompletion() {
        return done ? "1" : "0";
    }
```

Commit:
Add Gradle support

howardwhw2 committed:
Level 1

howardwhw2 committed:
added level 2

howardwhw2 committed:
level 3

howardwhw2 committed:
level 4

howardwhw2 committed:
added testing

howardwhw2 committed:
level 5

howardwhw2 committed:
level 6

howardwhw2 committed:
add Level-7

howardwhw2 committed:
add Level-8

howardwhw2 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Van.java

howardwhw2 committed:
Add OOP and package

howardwhw2 committed:
no message

howardwhw2 committed:
Revert "no message"

This reverts commit 3a9412facb4b5bbaafccf77dbdaa9c01d4ddc340.

howardwhw2 committed:
no message

howardwhw2 committed:
Merge branch 'master' into branch-Level-7

howardwhw2 committed:
Add 3 Junit tests

howardwhw2 committed:
Merge branch 'master' into branch-Level-7

howardwhw2 committed:
Merge branch 'master' of https://github.com/howardwhw2/ip

howardwhw2 committed:
Merge branch 'branch-Level-7'

howardwhw2 committed:
Add JavaDocs to 11 classes

howardwhw2 committed:
Edit code to follow coding standard

howardwhw2 committed:
Add find command(level-9)

howardwhw2 committed:
Merge branch 'branch-A-CodingStandard'

howardwhw2 committed:
Merge branch 'branch-Level-9'

howardwhw2 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

howardwhw2 committed:
Add gradle support and recitfied coding standard violations

howardwhw2 committed:
Add level-10

MelvinCZJ commented on src/main/java/van/FindCommand.java:
```diff
@@ -0,0 +1,44 @@
+/**
+ * Abstracts commands that finds particular keywords in existing tasks
+ */
+public class FindCommand implements Command {
```
Great naming convention! 👍

MelvinCZJ commented on src/main/java/van/FindCommand.java:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    /**
+     * Calls the methods needed in order to find a keyword
```
I think your JavaDocs are informative yet concise! 😄

MelvinCZJ commented on src/main/java/van/Parser.java:
```diff
@@ -88,6 +88,12 @@ public static Command parseCommand(String parameter) {
             output = new InvalidCommand("Please use integer numbers e.g. 1, 2");
```
Would the use of the word "integer" be easily understood by everyone?

MelvinCZJ commented on src/main/java/van/Task.java:
```diff
@@ -28,4 +28,13 @@ public void setDone() {
     public void setunDone() {
```
Shouldn't the name be setUndone instead? (camelcase)

MelvinCZJ commented on src/main/java/van/Ui.java:
```diff
@@ -46,4 +46,12 @@ public void printList (ArrayList<Task> taskList) {
   }
+
+  public void printResults (ArrayList<Task> taskList, String keyword) {
+    this.printDivider();
```
Could the printing be handled by another class instead?

MelvinCZJ commented on src/main/java/van/Ui.java:
```diff
@@ -46,4 +46,12 @@ public void printList (ArrayList<Task> taskList) {
       System.out.println(i+1 + ". " + taskList.get(i).getStatus());
```
There should be a single space before and after the operators!

MelvinCZJ reviewed (commented):
Overall, clean and very readable code. Apart from a few coding style violations, no other problems! Great job! 👍

howardwhw2 committed:
Add message for mark/unmark command and GUI improvement

howardwhw2 committed:
Add Asserts

howardwhw2 committed:
Merge pull request #1 from howardwhw2/master

Add message for mark/unmark command and GUI improvement

howardwhw2 committed:
Improve code quality

There are some methods that are too long which makes it harder to find errors and debug

Refactoring the code would make the code quality better
and make it easier for future changes and improvements

Thus, the code will be be refactored by moving repeated code and breaking
 long methods into samller methods to make the code quality better

howardwhw2 committed:
Merge pull request #2 from howardwhw2/branch-A-CodeQuality

Improve code quality

howardwhw2 committed:
Add tag extension

There is a lack of any features to categorize tasks in the list to make it
easier to see what the task is about

Adding of tag a tag feature would help to solve this problem

To do so adding the function to be able to tag tasks with additional details would
make it easier to organize and categorize tasks.

howardwhw2 committed:
Add user guide and output to delete commands

howardwhw2 committed:
Add ui preview and various code quality and ui dialoguebox improvementts

howardwhw2 committed:
Rectify problem with file name

howardwhw2 committed:
Delete Ui.PNG to rectify file name

howardwhw2 committed:
Reupload Ui image

howardwhw2 committed:
Add additional Junit tests

howardwhw2 committed:
Fix minor typos
